openedx/core/djangoapps/django_comment_common/comment_client/user.py

```diff
@@ -144,57 +229,75 @@ def _retrieve(self, *args, **kwargs):
         url = self.url(action='get', params=self.attributes)
         retrieve_params = self.default_retrieve_params.copy()
         retrieve_params.update(kwargs)
-        if self.attributes.get('course_id'):
-            retrieve_params['course_id'] = str(self.course_id)
+        course_id = retrieve_params.get("course_id") or self.attributes.get("course_id")
+        if isinstance(course_id, CourseKey):
+            retrieve_params["course_id"] = str(course_id)
         if self.attributes.get('group_id'):
-            retrieve_params['group_id'] = self.group_id
-        try:
-            response = utils.perform_request(
-                'get',
-                url,
-                retrieve_params,
-                metric_action='model.retrieve',
-                metric_tags=self._metric_tags,
-            )
-        except utils.CommentClientRequestError as e:
-            if e.status_code == 404:
-                # attempt to gracefully recover from a previous failure
-                # to sync this user to the comments service.
-                self.save()
+            retrieve_params['group_id'] = self.attributes["group_id"]
+        course_key = utils.get_course_key(course_id)
+        if is_forum_v2_enabled(course_key):
+            if not retrieve_params.get("course_id"):
+                retrieve_params["course_id"] = str(course_key)
+            try:
+                response = forum_api.get_user(self.attributes["id"], retrieve_params)
+            except ForumV2RequestError as e:
+                self.save({"course_key": course_key})
+                response = forum_api.get_user(self.attributes["id"], retrieve_params)
```

Comment:
This was causing an error on my computer, where `self.attributes["id"]` was None. Strangely, the error has resolved itself.

What is this strange pattern of catching an exception to run the same `get_user` function one more time?

Reply:
actually we followed the same pattern how edx-platform was calling the V1 APIs before.
At first, it tries to get the user, if it doesn't exists. It then creates the user in forum APP and then again gets.
For the new user that goes to discussion MFE, this error always occur once.

Reply:
Got it! Sorry!